File: Backend/models/alarm.go
package models

import (
	"time"
)

type Alarm struct {
	ID           uint       `gorm:"primaryKey"`
	DeviceID     string     `gorm:"type:text;not null"`
	ScheduleTime *time.Time `gorm:"not null"`
	IsEnabled    bool       `gorm:"default:true"`
	DurationOn   int        `gorm:"default:5"` // in minutes

	RepeatType int `gorm:"type:int;default:1"` // 1=once,2=daily,3=weekly

	CreatedAt time.Time `gorm:"default:now()"`
	UpdatedAt time.Time `gorm:"default:now()"`

	Device Device `gorm:"foreignKey:DeviceID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}

File: frontend/lib/domain/entities/alarm.dart
class AlarmEntity {
  final String? id;
  final String? deviceId;
  final bool? isExecuted;
  final DateTime? scheduleTime;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const AlarmEntity({
    this.id,
    this.deviceId,
    this.isExecuted,
    this.scheduleTime,
    this.createdAt,
    this.updatedAt,
  });
}
